src/Bicep.Core/Syntax/FunctionCallSyntax.cs

```diff
@@ -7,13 +7,13 @@ namespace Bicep.Core.Syntax
 {
     public class FunctionCallSyntax : FunctionCallSyntaxBase
     {
-        public FunctionCallSyntax(IdentifierSyntax name, Token openParen, IEnumerable<SyntaxBase> children, Token closeParen)
+        public FunctionCallSyntax(IdentifierSyntax name, Token openParen, IEnumerable<SyntaxBase> children, Token? closeParen)
```

Comment:
Do we have other examples like this of using null tokens to represent un-parsed stuff? Is there a good reason to do it this way?

Elsewhere I believe the approach we take is to insert skipped syntax - this makes it easy to associate diagnostics. I'm also keen to avoid 2 patterns of doing a similar thing.

Reply:
We use nullable `Token`s to represent missing _optional_ stuff, but you're right that it's not a pattern we have previously used for missing required syntax elements.

I had originally used [`SyntaxBase`](https://github.com/Azure/bicep/pull/16508/commits/db279d53e86341ef0fe3f622970d156e49c2c608#diff-115ab8c2137e75e35c1785215471c7a0d4c20613de5a4ad800d99fd38d9e15a9) and then switched to the nullable token to keep the "missing closing ')'" diagnostic attached to the correct line, but I think that's been mooted by the last commit since you convinced me that we don't need to recover from unclosed parens in the middle of a file. Will revert.